File: app/src/main/java/com.github.anrimian.musicplayer/di/app/play_list/PlayListComponent.java
package com.github.anrimian.musicplayer.di.app.play_list;

import com.github.anrimian.musicplayer.ui.playlist_screens.playlist.PlayListPresenter;
import com.github.anrimian.musicplayer.ui.playlist_screens.rename.RenamePlayListPresenter;

import dagger.Subcomponent;

@Subcomponent(modules = PlayListModule.class)
public interface PlayListComponent {

    PlayListPresenter playListPresenter();
    RenamePlayListPresenter changePlayListPresenter();

}

File: app/src/main/java/com/github/anrimian/musicplayer/di/app/play_list/PlayListComponent.kt
package com.github.anrimian.musicplayer.di.app.play_list


import com.github.anrimian.musicplayer.ui.playlists.rename.RenamePlayListPresenter
import dagger.Subcomponent

@Subcomponent(modules = [ PlayListModule::class ])
interface PlayListComponent {

    fun changePlayListPresenter(): RenamePlayListPresenter

}
